plugins/default/lib/mongodb_health.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+import argparse
+import sys
+import traceback
+from collections import Counter
+
+import openshift
+import nagios
+from mongodb import ReplsetStatuses
+
+
+def generate_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-s', '--selector', type=str, action="store",
+                        required=True, help="Label Selector")
+    return parser
+
+
+check_mongodb_cmd = ("bash", "-c",
+                     "mongo 127.0.0.1/admin "
+                     "-u \"admin\" -p \"$MONGODB_ADMIN_PASSWORD\" "
+                     "--eval=\"rs.status().myState\" --quiet")
+
+
+def parse_mongo_result(output):
+    try:
+        return int(output.rstrip("\n"))
+    except:
+        return None
+
+
+def analize(status):
+    if status is None:
+        return nagios.UNKNOWN
+    elif status == ReplsetStatuses.PRIMARY or status == ReplsetStatuses.SECONDARY:
+        return nagios.OK
+    elif status == ReplsetStatuses.ARBITER or status == ReplsetStatuses.STARTUP:
+        return nagios.WARN
+    else:
+        return nagios.CRIT
+
+
+def report(pods, rs_statuses, nag_statuses):
+    if len(filter(None, pods)) < 1:
+        print "%s: Unable to locate any pods running mongodb" % (
+            nagios.status_code_to_label(nagios.UNKNOWN),)
+        return nagios.UNKNOWN
+
+    unique_rs_statuses = Counter(rs_status for rs_status in rs_statuses)
+    unique_nag_statuses = Counter(nag_status for nag_status in nag_statuses)
+
+    ret = max(unique_nag_statuses)
+
+    if unique_rs_statuses[ReplsetStatuses.PRIMARY] != 1:
+        ret = nagios.CRIT
+        print "%s: There are %s nodes claiming to have the primary replica set" % (
+            nagios.status_code_to_label(ret), unique_rs_statuses[ReplsetStatuses.PRIMARY])
+    elif len(rs_statuses) % 2 == 0:
+        ret = nagios.WARN
+        print "%s: There are an even number of voting members (%s) partipating in the replica set" % (
+            nagios.status_code_to_label(ret), unique_rs_statuses[ReplsetStatuses.PRIMARY])
+    else:
+        print "%s: There are %s primary and %s secondary members in the replica set" % (
+            (nagios.status_code_to_label(ret)),
+            unique_rs_statuses[ReplsetStatuses.PRIMARY],
+            unique_rs_statuses[ReplsetStatuses.SECONDARY])
+
+    for pod, rs_status, nag_status in zip(pods, rs_statuses, nag_statuses):
+        print "%s: %s - %s" % (
+            nagios.status_code_to_label(nag_status),
+            pod,
+            ReplsetStatuses.reverse_mapping[rs_status])
+
+    return ret
+
+
+def check(selector):
+    project = openshift.get_project()
+
+    pods = openshift.get_running_pod_names(project, "name="+selector)
```

Comment:
`"name="` should not be hard coded here, it is part of the selector. A selector has the form `"foo=bar"`.

I suggest using a keyword argument for clarity:

```python
pods = openshift.get_running_pod_names(project, selector=selector)
```

As we talked on IRC, if there would be other flags that we want to pass to `get_running_pod_names`, we'd add in more keyword arguments, instead of relying on positional argument order.